### PR TITLE
Merge unreleased/rs3 to release/1.7

### PR DESCRIPTION
**Pull request by boingoing**

### CODE REVIEW

File: lib/Common/ChakraCoreVersion.h [hunk 1]
```diff
@@ -54,7 +54,7 @@
 // * Does not add anything to the file description
 
 // ChakraCore RELEASE and PRERELEASE flags
-#define CHAKRA_CORE_VERSION_RELEASE 1
```
Comment (MSLaguana) [inline lib/Common/ChakraCoreVersion.h]: Why did you change this? My understanding is that it is set to 1 since 1.7 is a release branch?

Comment (boingoing) [reply lib/Common/ChakraCoreVersion.h]: You are correct, this was an oversight.

**Comment by dilijev** [comment]
Commenting for the record that the size of this diff is mostly line ending normalizations to test files:

With line ending changes:

```
>git diff --shortstat b00660633
 347 files changed, 690861 insertions(+), 689686 deletions(-)
```

Without line ending changes:

```
>git diff --shortstat -w b00660633
 347 files changed, 1637 insertions(+), 462 deletions(-)
```

File: .gitattributes [hunk 2]
```diff
@@ -3,3 +3,9 @@
 test/**/*.js -crlf
 test/es6/HTMLComments.js binary diff=cpp
 *.wasm binary
+*.cpp text eol=lf diff=cpp
+*.h text eol=lf diff=cpp
+*.inl text eol=lf diff=cpp
+*.vcproj text eol=crlf diff=xml
+*.vcxproj text eol=crlf diff=xml
```
Comment (dilijev) [inline .gitattributes]: so far, AFAIK, we've been treating these files as LF with no ill effects.

File: .gitattributes [hunk 3]
```diff
@@ -3,3 +3,9 @@
...
+*.sln text eol=crlf diff=xml
```
Comment (dilijev) [inline .gitattributes]: >*.sln text eol=crlf [](start = 0, length = 19)

yes, AFAIK, sln requires CRLF

File: lib/Backend/LowerMDShared.cpp [hunk 4]
```diff
@@ -6306,6 +6306,7 @@ LowererMD::EmitLoadFloatCommon(IR::Opnd *dst, IR::Opnd *src, IR::Instr *insertIn
     return labelDone;
 }
 
+
```
Comment (dilijev) [inline lib/Backend/LowerMDShared.cpp]: nit: added unnecessary newline

File: lib/Runtime/ByteCode/ByteCodeCacheReleaseFileVersion.h [hunk 5]
```diff
@@ -4,6 +4,6 @@
 //-------------------------------------------------------------------------------------------------------
 // NOTE: If there is a merge conflict the correct fix is to make a new GUID.
 
-// {C6C1CBF8-B232-4E4C-8681-34E404F78F16}
+// {F856871A-E536-4CA1-83E6-819E26C7DC99}
```
Comment (dilijev) [inline lib/Runtime/ByteCode/ByteCodeCacheReleaseFileVersion.h]: Just a note (in case it wasn't already done this way) that if there is a merge conflict in this file, please generate a new GUID.

File: lib/Runtime/Language/DynamicProfileInfo.cpp [hunk 6]
```diff
@@ -2460,3 +2460,4 @@ const char* GetBailOutKindName(IR::BailOutKind kind)
 }
 #endif
 #endif
+
```
Comment (dilijev) [inline lib/Runtime/Language/DynamicProfileInfo.cpp]: nit: unnecessary extra line

File: lib/Runtime/Language/DynamicProfileInfo.h [hunk 7]
```diff
@@ -972,3 +972,4 @@ namespace Js
 #endif
 };
 #endif
+
```
Comment (dilijev) [inline lib/Runtime/Language/DynamicProfileInfo.h]: nit: unnecessary extra line

**Comment by dilijev** [comment]
LGTM but I question the need to take these line ending changes in the test files. I'd say we re-normalize them to what they were before unless there's a good reason to change.

/cc @tcare

**Comment by Penguinwizzard** [comment]
The issue with the test files is that they're currently specified as "-crlf"; this doesn't mean "not crlf", but means "-text", and as such disables any git normalization and has them explicitly use the user's platform's eol normalization. We should likely have them set as "text eol=lf".

File: lib/Runtime/ByteCode/ByteCodeCacheReleaseFileVersion.h (hunk 5 above)
Comment (boingoing) [reply lib/Runtime/ByteCode/ByteCodeCacheReleaseFileVersion.h]: This is a new GUID. 👍

File: .gitattributes (hunk 2 above)
Comment (tcare) [reply .gitattributes]: +1, where did this change originate from?

**Comment by tcare** [comment]
Changing line endings should not be a part of this PR. Let's address this during downtime in December along with other invasive .gitattributes changes I want to make.

File: .gitattributes (hunk 2 above)
Comment (dilijev) [reply .gitattributes]: Sounds like @Penguinwizzard recommended this spec as a way to help git deal with merge conflicts on line endings -- and it possibly had the opposite from the intended effect. :P

Comment (tcare) [reply .gitattributes]: There's an easy way to strip line ending changes from a PR. I'll have to dig it up from email somewhere.

**Comment by tcare** [comment]
Do we have a list of changes for this PR?

**Comment by dilijev** [comment]
@Penguinwizzard as always, I'll add that because some of our (especially parser-related) JS tests are sensitive to line endings and other whitespace, a blanket policy for line endings in JS files might be ill-advised. However, OTOH I'm not sure why we never bothered to move all such files to their own directory and make a specific policy for that.

**Comment by tcare** [comment]
@dilijev is right, normalizing line endings in .js files will introduce false negatives for scanner tests.

**Comment by dilijev** [comment]
> introduce false negatives

The reason everyone is skittish about improving our test code is we don't when a change might cause the test to stop covering the intended code path. Too bad we don't have a way to determine that a test is testing the right things, and to protect against test coverage regression. (Code coverage isn't quite the same thing.)
